### PR TITLE
Add Support For Dimension Engineering Sabertooth Motor Controller

### PR DESCRIPTION
**Pull request by thegreatco**

### CODE REVIEW

File: components/motor/dimensionengineering/sabertooth.go [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+	return errors.New("not supported")
+}
+
+// IsPowered returns if the motor is currently on or off.
+func (m *Motor) IsPowered(ctx context.Context, extra map[string]interface{}) (bool, error) {
+	return m.jogging, nil
+}
+
+// Config adds DimensionEngineering-specific config options.
+type Config struct {
```
Comment (randhid) [inline components/motor/dimensionengineering/sabertooth.go:65]: [nit] add `...,omitempty"` for unneeded fields like here:
https://github.com/viamrobotics/rdk/blob/0e659a0723aeb3d8b0b3038a2e7b1735732305ba/components/gantry/oneaxis/oneaxis.go#L35

and check them with a `Validate` function like here: 
https://github.com/viamrobotics/rdk/blob/0e659a0723aeb3d8b0b3038a2e7b1735732305ba/components/movementsensor/imuvectornav/imu.go#L37

File: components/motor/dimensionengineering/sabertooth.go [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+type PinConfig struct {
+	EStop string `json:"EStop"`
+}
+
+func init() {
+	controllers = make(map[string]*controller)
+
+	_motor := registry.Component{
+		Constructor: func(ctx context.Context, _ registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
+			return NewMotor(ctx, config.ConvertedAttributes.(*Config), logger)
```
Comment (randhid) [inline components/motor/dimensionengineering/sabertooth.go:84]: [nit] check if config has successfully been converted and is ok like here:
https://github.com/viamrobotics/rdk/blob/0e659a0723aeb3d8b0b3038a2e7b1735732305ba/components/gantry/oneaxis/oneaxis.go#L145

File: components/motor/dimensionengineering/sabertooth.go [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+	ctrl.serialDevice = c.SerialDevice
+	ctrl.logger = logger
+	ctrl.address = c.Address
+
+	if c.TestChan != nil {
+		ctrl.testChan = c.TestChan
+	} else {
+		serialOptions := serial.OpenOptions{
+			PortName:          c.SerialDevice,
+			BaudRate:          9600,
```
Comment (randhid) [inline components/motor/dimensionengineering/sabertooth.go:208]: Will baudrate ever be configurable?

File: components/motor/dimensionengineering/sabertooth.go [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+	}
+	switch name {
+	default:
+		return nil, fmt.Errorf("no such command: %s", name)
+	}
+}
+
+// Properties returns the additional features supported by this motor.
+func (m *Motor) Properties(ctx context.Context, extra map[string]interface{}) (map[motor.Feature]bool, error) {
+	return map[motor.Feature]bool{}, nil
```
Comment (randhid) [inline components/motor/dimensionengineering/sabertooth.go:348]: `PositionReporting : false`

File: components/motor/dimensionengineering/sabertooth.go [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+)
+
+const (
+	// https://www.dimensionengineering.com/datasheets/Sabertooth2x60.pdf
+	modelName = "sabertooth"
```
Comment (randhid) [inline components/motor/dimensionengineering/sabertooth.go:25]: `de-sabertooth`? or something like that

**Review by randhid — CHANGES_REQUESTED** [review]

**Comment by randhid** [comment]
Few tiny cleanup nits then I can approve.

File: components/motor/dimensionengineering/sabertooth.go (hunk 3 above)
Comment (thegreatco) [reply components/motor/dimensionengineering/sabertooth.go:208]: It is possible to change it, that's why `common.go` has the `baud2400`, `baud9600`, etc. Is there a way to express the desire to change the baudrate in viam?

Comment (randhid) [reply components/motor/dimensionengineering/sabertooth.go:222]: same as you've used the config for serial device for port name, you can have a c.BaudRate field and you can set it through config fields in app. `movementsensor/gps-nmea` has an example of this as I recall.

You can also check if it's not set, and se a default (9600) and log  it as info that way.

**Review by randhid — APPROVED** [review]

**Pull request merged**